### ISSUE TITLE
Reconnect in packages

### ISSUE THREAD

**Issue by IllusionPerdu** [xmppjs/xmpp.js#507]
Hello,
With the recent change I think in the xmpp.js/packages/reconnect/index.js the 

```
module.exports = function reconnect(entity) {
  const r = new Reconnect(entity)
  r.start()
  return r
}
```
should be 
```
module.exports = function reconnect(entity) {
  const r = new Reconnect(entity)
  r.start()
  entity.reconnect = r
  return r
}
```
without this we don't make action on reconnected.

Sorry for my bad english :s

**Comment by sonnyp** [comment]
@IllusionPerdu pas sûr de comprendre, reconnect ne marche plus pour toi sur master?

**Comment by IllusionPerdu** [comment]
Actuellement j'essaie de faire un client web.
Avant qu'il y est les modifications de la mise en place de reconnect en package vu que j'ai besoin de faire une action sur la reconnection je faisais : 
```
client.plugins.reconnect.delay = 10000;
client.plugins.reconnect.on('reconnected', function () {
...
});
```
mais maintenant ce n'est plus exposé donc plus possible de faire une action sur la reconnection. Avec la modification que je propose ça permet de pouvoir redéfinir le delay et aussi de pouvoir agir en cas de reconnection avec un 
```
client.reconnect.delay = 10000;
client.reconnect.on('reconnected', function () {
...
});
```
Ou faut peux être les ré-exposer dans les plugins... Après j'ai peut-être louper un truc ;) 

En tout cas bravo pour cette lib qui contrairement à strophe.js permet d'être mise dans un worker / sharedworker ;) et qui en plus est beaucoup plus petite... Après il manque cruellement de la documentation mais je suppose que c'est le temps de la réécriture et que ça viendra par la suite.

**Comment by sonnyp** [comment]
> En tout cas bravo pour cette lib qui contrairement à strophe.js permet d'être mise dans un worker / sharedworker ;) et qui en plus est beaucoup plus petite...

Merci ça fait plaisir

> Après il manque cruellement de la documentation mais je suppose que c'est le temps de la réécriture et que ça viendra par la suite.

Oui, surtout que master est encore un peu en chamboulement mais ça devrait plus trop bouger.

C'est exposé différement, le système de plugins était bizarre je suis en train de modifier ça, il ne reste plus que quelques plugins.

```js
const {xmpp, xml, jid} = require('@xmpp/client')

const {client, reconnect} = xmpp()

client.start(...)

reconnect.delay = 10000;
reconnect.on('reconnected', function () {
...
});
```

**Comment by sonnyp** [comment]
voir https://github.com/xmppjs/xmpp.js/blob/master/packages/client/index.js

**Comment by IllusionPerdu** [comment]
Merci pour ta réponse, j'essayerai un peu plus tard (et oui c'est la fonctionnalité galère à tester). Je me doutais un peu que j'avais loupé un truc ;) Il va vraiment falloir que je me mettent a étudier comment browserify fonctionne vu que je suis de la vielle école ;) Il faut bien avouer que c'est dernière années le javascript c'est bien enrichie.

ça pourrais être intéressant d'enrichir l'exemple... et mettre une petite note sur le readme du reconnect.

**Comment by sonnyp** [comment]
> ça pourrais être intéressant d'enrichir l'exemple... et mettre une petite note sur le readme du reconnect.

Absolument, la doc viendra quand j'aurais terminé la migration des plugins

**Comment by IllusionPerdu** [comment]
Bon malheureusement, je confirme que ça ne fonctionne pas (tester sous google chrome).
Avec le `const {client, reconnect, ...} = xmpp()` le reconnect est undefined

en modifiant dans https://github.com/xmppjs/xmpp.js/blob/master/packages/client/index.js la fonction xmpp()
```javascript
function xmpp() {
  const client = new _$Client_18()
  var tmp = {};
  entries(packages)
    // Ignore browserify stubs
    .filter(([, v]) => typeof v === 'function')
    .map(function([k, v]) {
      tmp[k] = v(client)
    });
  return Object.assign(
    {client},
    tmp
  );
}
```
Tout semble fonctionner.

**Comment by sonnyp** [comment]
@IllusionPerdu oops, ok je regarde après le tâf
